src/target/ext/host_io.rs

```diff
@@ -0,0 +1,43 @@
+//! Provide Host I/O operations for the target.
+use crate::arch::Arch;
+use crate::common::{HostMode, HostOpenFlags};
+use crate::target::Target;
+
+/// An interface to send pread data back to the GDB client.
+pub struct PreadOutput<'a> {
+    cb: &'a mut dyn FnMut(&[u8]),
+}
+
+impl<'a> PreadOutput<'a> {
+    pub(crate) fn new(cb: &'a mut dyn FnMut(&[u8])) -> Self {
+        Self { cb }
+    }
+
+    /// Write out raw file bytes to the GDB debugger.
+    pub fn write(&mut self, buf: &[u8]) {
+        (self.cb)(buf)
+    }
+}
+
+/// Target Extension - Perform I/O operations on host
+pub trait HostIo: Target {
+    /// Open a file at filename and return a file descriptor for it, or return
+    /// -1 if an error occurs.
+    fn open(&self, filename: &[u8], flags: HostOpenFlags, mode: HostMode) -> i64;
+    /// Close the open file corresponding to fd and return 0, or -1 if an error
+    /// occurs.
+    fn close(&self, fd: usize) -> i64;
+    /// Read data from the open file corresponding to fd.
```

Comment:
Please copy the full documentation for the method from the vFile docs. It includes useful information for implementors, such as the fact that "The target may read fewer bytes; common reasons include packet size limits and an end-of-file condition"

In fact, you should probably copy the docs verbatim for all the other vFile methods as well (omitting any bits that don't apply at this level of abstraction, such as error-code handling)